sql/analyzer/indexed_joins.go

```diff
@@ -681,7 +681,17 @@ func getJoinIndex(
 	exprsByTable := joinExprsByTable(joinCondPredicates)
 	indexesByTable := make(joinIndexesByTable)
 	for table, cols := range exprsByTable {
-		idx := ia.IndexByExpression(ctx, ctx.GetCurrentDatabase(), normalizeExpressions(tableAliases, extractExpressions(cols)...)...)
+		exprs := extractExpressions(cols)
+		idx := ia.IndexByExpression(ctx, ctx.GetCurrentDatabase(), normalizeExpressions(tableAliases, exprs...)...)
+		// if we do not find a perfect index, take the first partial index if there is one.
```

Comment:
This only works for a single column index, won't match multiple columns. Add a todo?